### PR TITLE
Add CI tests and update to newer ACTS

### PR DESCRIPTION
**Pull request by Zehvogel**
BEGINRELEASENOTES
- Update CI to include tests builds and geometry loading
- Update CMake files and code to work with newer ACTS versions

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
Can you add pre-commit and license while you are at it?

**Comment by andresailer** [comment]
@paulgessinger Are you OK with the Apache 2 License?

File: k4ActsTracking/CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ gaudi_add_module(k4ActsTrackingPlugins
   EDM4HEP::edm4hep
   DD4hep::DDCore DD4hep::DDRec
   ActsCore ActsPluginDD4hep
-  ActsDD4hep::ActsDD4hep
+  #ActsDD4hep::ActsDD4hep
```
Comment (paulgessinger) [inline k4ActsTracking/CMakeLists.txt:10]: This shouldn't be needed anymore at all I don't think.
```suggestion
```

**Comment by Zehvogel** [comment]
Build test is waiting for https://github.com/key4hep/key4hep-spack/pull/522 as Acts is unfortunately built without the dd4hep plugin in key4hep atm..
```
[lreichen@pcphsft121 acts]$ ll /cvmfs/sw-nightlies.hsf.org/key4hep/releases/2023-06-24/x86_64-centos7-gcc12.2.0-opt/acts/23.2.1-rk7msr/include/Acts/
total 42
-rw-r--r--. 1 cvmfs cvmfs 1036 Jun 25 08:46 ActsVersion.hpp
drwxr-sr-x. 2 cvmfs cvmfs   58 Jun 25 08:48 Clusterization
drwxr-sr-x. 2 cvmfs cvmfs  113 Jun 25 08:48 Definitions
drwxr-sr-x. 2 cvmfs cvmfs  217 Jun 25 08:48 Digitization
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 EventData
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 Geometry
drwxr-sr-x. 3 cvmfs cvmfs  248 Jun 25 08:48 MagneticField
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 Material
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 Propagator
drwxr-sr-x. 2 cvmfs cvmfs 4096 Jun 25 08:48 Seeding
drwxr-sr-x. 3 cvmfs cvmfs  120 Jun 25 08:48 SpacePointFormation
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 Surfaces
drwxr-sr-x. 2 cvmfs cvmfs  153 Jun 25 08:48 TrackFinding
drwxr-sr-x. 3 cvmfs cvmfs  282 Jun 25 08:48 TrackFitting
drwxr-sr-x. 3 cvmfs cvmfs 4096 Jun 25 08:48 Utilities
drwxr-sr-x. 2 cvmfs cvmfs 4096 Jun 25 08:48 Vertexing
drwxr-sr-x. 3 cvmfs cvmfs  183 Jun 25 08:48 Visualization
```

**Comment by paulgessinger** [comment]
The other PR is merged but it seems like it's still failing to find the the plugin?

**Comment by Zehvogel** [comment]
We only expect the nightlies to pass and they do now. It will take until the next release for the other one...

**Comment by Zehvogel** [comment]
currently waiting for the opendatadetector to become available in the nightlies, it should be there tomorrow. Then we can use it directly in the ci without building it ourselves every time

**Comment by paulgessinger** [comment]
@Zehvogel I see, where is this built then? I'm wondering if keeping this in sync is really worth it since building it does not take a long time normally.

**Comment by Zehvogel** [comment]
> @Zehvogel I see, where is this built then? I'm wondering if keeping this in sync is really worth it since building it does not take a long time normally.

@paulgessinger Is your question where the ODD is built? We [added it to the key4hep-stack](https://github.com/key4hep/key4hep-spack/blob/main/packages/key4hep-stack/package.py#L97). At the moment we build the last tagged version, but we could also add it properly to the key4hep nightlies and have the latest main all the time...

**Pull request merged**